File: BE/src/main/java/com/example/Fstock/service/Service/ProductService.java
package com.example.Fstock.service.Service;

import com.example.Fstock.dto.request.CreateProductRequest;
import com.example.Fstock.dto.response.ColorInfo;
import com.example.Fstock.dto.response.ProductDto;
import org.apache.coyote.BadRequestException;
import org.springframework.data.domain.Page;
import org.springframework.web.multipart.MultipartFile;

import java.io.IOException;
import java.util.List;
import java.util.Set;

public interface ProductService  {
    ProductDto getProductById(int id);
    Page<ProductDto> getAllProducts(int pageNumber , int pageSize , String search);
    Page<ProductDto> getProductByFilter(Integer priceMin, Integer priceMax, String colorName, String sizeName, Integer categoryId, int pageNumber);
    Integer getQuantityInStock(int productId, String colorName, String sizeName);
    List<ColorInfo> getAllColor();
    List<String> getAllSize();
    void createProduct(CreateProductRequest createProductRequest) throws IOException;
    void deleteProduct(int productId);
    void updateProduct(int productId, CreateProductRequest createProductRequest) throws IOException;


}

File: FE/src/service/ProductService.js
import api from "../api/api";

export const getAllProducts = (pageNumber, pageSize , search) => {
    return api.get(`/products/page`, {
        params: {
            pageNumber: pageNumber,
            pageSize: pageSize,
            search: search
        }
    }); 
}

export const getProductById = (id) => {
    return api.get(`/products/${id}`); 
}

export const getProductByFilter = (filter) => {
    return api.get(`/products/filter`, {
        params: {
            priceMin : filter.priceMin,
            priceMax : filter.priceMax,
            category : filter.category,
            size : filter.size,
            color : filter.color,
            pageNumber : filter.pageNumber
        }
    }); 
}

export const getStockQuantity = (data) => {
    return api.get(`/products/quantity`, {
        params : {
            productId : data.productId,
            sizeName : data.sizeName,
            colorName : data.colorName
        }

    }); 
}
export const createProduct = (data) => {
    return api.post(`/products/create-product`, data , {
        headers: {
            'Content-Type': 'multipart/form-data'
        }
    }); 
}

export const updateProduct = (data , id) => {
    return api.put(`/products/update-product/${id}`, data , {
        headers: {
            'Content-Type': 'multipart/form-data'
        }
    }); 
}
export const deleteProduct = (id) => {
    return api.delete(`/products/delete/${id}`)
}


export const getColorNames = () => {
    return api.get(`/products/colors`); 
}
export const getSizeNames = () => {
    return api.get(`/products/sizes`); 
}
